### PR TITLE
Added Provider-Specific timeouts

### PR DESCRIPTION
**Pull request by jannefleischer**
For calculation of big Matrixes on our custom ORS-Server I needed more calculation time, so I included a new provider-specific timeout for orstools-qgis-plugin.

It is still a little bit ugly as I use some unnecessary recasting int/string for seconds-values...

### CODE REVIEW

File: ORStools/common/networkaccessmanager.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ class NetworkAccessManager(object):
             'exception' - the exception returne dduring execution
     """
 
-    def __init__(self, authid=None, disable_ssl_certificate_validation=False, exception_class=None, debug=True):
+    def __init__(self, authid=None, disable_ssl_certificate_validation=False, exception_class=None, debug=True, timeout=None):
```
Comment (koebi) [inline ORStools/common/networkaccessmanager.py:145]: default to 60 here should work :)

**Review by koebi — CHANGES_REQUESTED** [review]
Hey,

thank you very much for your pull request, it is greatly appreciated :))
Sorry for the long silence from our side :(

I think sticking to a default of 60 seconds as a timeout is great, but should be consistently used everywhere.
This could be made clear by setting a `placeholderText()` to `60` in `timeout_text`, such that users see that a default is set.
This default would be used by using `provider.get('timeout', 60)` rather than provider['timeout'].

I think this would simplify a lot of things :) 

Best regards,
Jakob



This would enable us to drop a bunch of `timeout`s in the code,

**Comment by koebi** [comment]
Hey @jannefleischer,

it's been over a month since the review, so I wanted to follow up on this :)
Given our response time, your inactivity is definitely excused :smile:
I just wanted to know whether you plan on working on this any time soon.

If you don't want to or simply don't have the time, just let us know so we can move forward with this :)

Best regards,
Jakob

**Comment by jannefleischer** [comment]
Sorry for the major lag in response time. At the moment I can't see when there is time once again for this. But currently I am overwhelmed by other work, so sadly, I can't promise anything :(

**Comment by koebi** [comment]
This has been reworked slightly and merged in 7ce0407b1925d39b5aa2961f24014f8067523451. Thank you very much for your work, glad to finally see this happen :)

**Pull request closed without merging**